torch/nativert/executor/Placement.h

```diff
@@ -0,0 +1,60 @@
+#pragma once
+
+#include <c10/core/Device.h>
+#include <c10/util/Logging.h>
+
+#include <optional>
+#include <unordered_map>
+
+namespace torch::nativert {
+
+/**
+ * @brief Normalizes a c10::Device by ensuring a canonical form.
+ *
+ * This function returns a normalized version of the input device:
+ * - For CPU devices, the returned device will have no index (i.e., the default CPU device).
+ * - For CUDA devices, if no index is specified, index 0 is assumed.
+ * - For other device types, the function will raise an error.
+ *
+ * @param device The input c10::Device to normalize.
+ * @return A normalized c10::Device with standardized indexing.
+ *
+ * @throws c10::Error If the device type is not CPU or CUDA.
+ */
+
+TORCH_API c10::Device normalizeDevice(const c10::Device& device);
+
+/**
+ * Returns true if the two devices are the same and has the same device index (if cuda).
+ */
+TORCH_API bool isSameDevice(
+    const c10::Device& device1,
+    const c10::Device& device2);
+
+
+/**
+ * @brief A utility class for managing device placement mappings.
+ *
+ * The Placement class provides a way to map source devices to target devices.
+ * It supports both explicit per-device mappings and a default device fallback.
+ * This is used for device parameterization in Torch Native Runtime, and to support
```

Comment:
What does "device parametrization" means?

Reply:
Since this is only to be used with the nativert API, I would say it the way you said it above: this is the argument taken in nativert to map from model artifact device to the device it should run on.